src/file/reader.rs

```diff
@@ -484,6 +520,16 @@ mod tests {
   use parquet_format::TypeDefinedOrder;
   use util::test_common::{get_temp_file, get_test_file, get_test_path};
 
+  #[test]
+  fn test_read_from_memory() {
```

Comment:
Test is a bit misleading, it should be something `test_read_from_memory_error_not_enough_bytes`, or something like that. I assume it fails because there are not enough bytes for the footer. IMHO, we should explicitly check the error.